akka-remote/src/main/java/akka/remote/artery/AbstractEntry.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package akka.remote.artery;
+
+// Copied from https://github.com/JCTools/JCTools/tree/master/jctools-experimental/src/main/java/org/jctools/maps
+
+import java.util.Map;
+
+/**
+ * A simple implementation of {@link java.util.Map.Entry}.
+ * Does not implement {@link java.util.Map.Entry.setValue}, that is done by users of the class.
+ *
+ * @since 1.5
+ * @author Cliff Click
+ * @param <TypeK> the type of keys maintained by this map
+ * @param <TypeV> the type of mapped values
+ */
+
+abstract class AbstractEntry<TypeK,TypeV> implements Map.Entry<TypeK,TypeV> {
```

Comment:
Yeah, for small maps just copying parts of an array to another place to keep it sorted and then do binary search is surprisingly fast. The question is the update frequency compared to queries. There are ever fancier array based layouts but I am not sure it is worth it.